x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -161,431 +162,560 @@ export const ApiKeyFlyout: FunctionComponent<ApiKeyFlyoutProps> = ({
-            />
-            {form.values.customPrivileges && (
+            {apiKey ? (
               <>
-                <EuiSpacer size="m" />
-                <EuiFormRow
-                  data-test-subj="apiKeysRoleDescriptorsCodeEditor"
-                  helpText={
-                    <DocLink
-                      app="elasticsearch"
-                      doc="security-api-create-api-key.html#security-api-create-api-key-request-body"
+                <EuiSpacer />
+                <EuiFlexGroup gutterSize="xl">
+                  <EuiFlexItem grow={false}>
+                    <EuiFormRow
+                      label={
+                        <FormattedMessage
+                          id="xpack.security.accountManagement.apiKeyFlyout.typeLabel"
+                          defaultMessage="Type"
+                        />
+                      }
                     >
-                      <FormattedMessage
-                        id="xpack.security.accountManagement.apiKeyFlyout.roleDescriptorsHelpText"
-                        defaultMessage="Learn how to structure role descriptors."
-                      />
-                    </DocLink>
+                      <ApiKeyBadge type={apiKey.type} />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <EuiFormRow
+                      label={
+                        <FormattedMessage
+                          id="xpack.security.accountManagement.apiKeyFlyout.ownerLabel"
+                          defaultMessage="Owner"
+                        />
+                      }
+                    >
+                      <UsernameWithIcon username={apiKey.username} />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <EuiFormRow
+                      label={
+                        <FormattedMessage
+                          id="xpack.security.accountManagement.apiKeyFlyout.createdLabel"
+                          defaultMessage="Created"
+                        />
+                      }
+                    >
+                      <TimeToolTip timestamp={apiKey.creation} />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <EuiFormRow
+                      label={
+                        <FormattedMessage
+                          id="xpack.security.accountManagement.apiKeyFlyout.statusLabel"
+                          defaultMessage="Status"
+                        />
+                      }
+                    >
+                      <ApiKeyStatus expiration={apiKey.expiration} />
+                    </EuiFormRow>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+                <EuiHorizontalRule />
+              </>
+            ) : canManageCrossClusterApiKeys ? (
```

Comment:
Nit: Would it be helpful to display something in place of type selection when the user cannot manage CC API keys? In other interfaces we favor disabling options that cannot be accessed, but I don't think that really works here either. Maybe just displaying the type like we do when editing an API key? The user will see that they created a personal key in the table, but there is zero indication that is what they're creating in the flyout.

Reply:
That's a really good point - I've added the badge to the form as suggested: 

<img width="588" alt="Screenshot 2023-08-04 at 16 06 47" src="https://github.com/elastic/kibana/assets/190132/44f35d9d-3f69-47b9-bbf5-1eeada64e7a8">